query/security/ldap/src/main/java/org/codice/ddf/admin/ldap/discover/LdapTestAttributeMappings.java

```diff
@@ -0,0 +1,161 @@
+import org.codice.ddf.admin.ldap.commons.LdapConnectionAttempt;
+import org.codice.ddf.admin.ldap.commons.LdapTestingUtils;
+import org.codice.ddf.admin.ldap.fields.LdapDistinguishedName;
+import org.codice.ddf.admin.ldap.fields.config.LdapSettingsField;
+import org.codice.ddf.admin.ldap.fields.connection.LdapBindUserInfo;
+import org.codice.ddf.admin.ldap.fields.connection.LdapConnectionField;
+import org.codice.ddf.admin.security.common.fields.wcpm.ClaimsMapEntry;
+import org.forgerock.opendj.ldap.Connection;
+import org.forgerock.opendj.ldap.Filter;
+import org.forgerock.opendj.ldap.SearchScope;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.ImmutableList;
+
+public class LdapTestAttributeMappings extends TestFunctionField {
+    private static final Logger LOGGER = LoggerFactory.getLogger(LdapTestAttributeMappings.class);
+
+    public static final String FIELD_NAME = "testAttributeFields";
+
+    public static final String DESCRIPTION =
+            "Tests whether the attributes mapped to claims exist on users beneath the user base.";
+
+    private LdapConnectionField conn;
+
+    private LdapBindUserInfo bindInfo;
+
+    private LdapSettingsField settings;
+
+    private LdapTestingUtils utils;
+
+    public LdapTestAttributeMappings() {
+        super(FIELD_NAME, DESCRIPTION);
+        conn = new LdapConnectionField().useDefaultRequired();
+        bindInfo = new LdapBindUserInfo().useDefaultRequired();
+        settings = new LdapSettingsField().useDefaultUserAttributes();
+        updateArgumentPaths();
+
+        utils = new LdapTestingUtils();
+    }
+
+    @Override
+    public List<DataType> getArguments() {
+        return ImmutableList.of(conn, bindInfo, settings);
+    }
+
+    @Override
+    public FunctionField<BooleanField> newInstance() {
+        return new LdapTestAttributeMappings();
+    }
+
+    @Override
+    public BooleanField performFunction() {
+        try (LdapConnectionAttempt connectionAttempt = utils.bindUserToLdapConnection(conn,
+                bindInfo)) {
+            addMessages(connectionAttempt);
+
+            if (containsErrorMsgs()) {
+                return new BooleanField(false);
+            }
+
+            Connection ldapConnection = connectionAttempt.result();
+
+            checkUserDirExists(ldapConnection);
+
+            // Short-circuit return here, if either the user or group directory does not exist
+            if (containsErrorMsgs()) {
+                return new BooleanField(false);
+            }
+
+            settings.attributeMapField()
+                    .getList()
+                    .stream()
+                    .map(ClaimsMapEntry::claimValueField)
+                    .filter(claim -> !mappingAttributeFound(ldapConnection, claim.getValue()))
+                    .forEach(claim -> addArgumentMessage(mappingAttributeNotFoundError(claim.path())));
+
+            if (containsErrorMsgs()) {
```

Comment:
do we need to do this check? looks like the last line of this method is doing this check.